### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by Bludidi**
# In this Pull Request I have implemented the following:
- Created a single-page application (SPA)  with React and Redux.
- Retrieved data from the API and stored in the Redux store.
- Filter data that is retrieved from the API using a filter stateless component.
- Each page has a unique route within the SPA.

## Home page
When the page loads, the web app shows the list of currencies that could be filtered out by currency name. Along with the currency name, it displays some numeric values per currency and other data from the API. When a currency is clicked, the application navigates to the details page and displays detailed data about that particular currency.

## Details page
On the details page, the web app retrieves data from the API to show detailed data of the currency. When the user clicks on the "Back" button (<), the user navigates to the home page.

## Demo video and live demo link
- [Presentation](https://www.loom.com/share/a892a52749db4a27914d059dce980620) 👈 click here
-  [crypto-home](https://hilarious-pasca-2d43e9.netlify.app) 👈 click here

### CODE REVIEW

File: src/__tests__/App.test.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { render } from '@testing-library/react';
```
Comment (SajjadAhmad14) [inline src/__tests__/App.test.js:1]: Please write unit tests components and for pure functions in Redux code.
Please write integration tests for app using React Testing Library.

tests

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Brian, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
README is professional. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/__tests__/App.test.js (hunk 1 above)
Comment (mahbubul14) [reply src/__tests__/App.test.js:1]: > Please write unit tests components and for pure functions in Redux code. Please write integration tests for the app using React Testing Library.
> 
> tests

- [ ] As the previous reviewer asked, kindly work on it. :+1:

**Review by mahbubul14 — CHANGES_REQUESTED** [review]
Hi @Bludidi,

Great work on making the changes requested by the previous reviewer 👏

You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

# Required Changes ♻️

- [ ] Kindly check the inline comments under the review. :arrow_down: 

###  To Highlights
:heavy_check_mark: No linter error.
:heavy_check_mark: Proper Github flow used.
:heavy_check_mark: Descriptive PR title & description.

### Optional suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

- [ ] The commit message should be written in the imperative tense and the first letter should also be capitalized. `Ex: "Fix bug" instead of "fix bug".` [For more details...](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also reach me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by arslanbisharat** [comment]
Hi,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**